eZ/Publish/API/Repository/Tests/ContentServiceTest.php

```diff
@@ -6186,4 +6188,115 @@ private function getExpectedMediaContentInfoProperties()
+            'folder',
+            $this->generateId('location', 2),
+            [
+                'name' => 'Folder US',
+            ]
+        );
+
+        $publishedContent = $contentService->publishVersion($contentDraft->versionInfo);
+
+        $deDraft = $contentService->createContentDraft($publishedContent->contentInfo);
+
+        $contentUpdateData = $this->translateField(
+            $deDraft,
+            'name',
+            new Value('Folder GER'),
+            'ger-DE'
+        );
+
+        $gerContent = $contentService->updateContent($deDraft->versionInfo, $contentUpdateData);
+
+        $updatedContent = $contentService->loadContent($gerContent->id, null, $gerContent->versionInfo->versionNo);
+        $this->assertEquals(
+            [
+                'eng-US' => 'Folder US',
+                'ger-DE' => 'Folder GER',
+            ],
+            $updatedContent->fields['name']
+        );
+
+        $frDraft = $contentService->createContentDraft($publishedContent->contentInfo);
+
+        $contentUpdateData = $this->translateField(
+            $frDraft,
+            'name',
+            new Value('Folder FR'),
+            'fre-FR'
+        );
+        $frContent = $contentService->updateContent($frDraft->versionInfo, $contentUpdateData);
+        $contentService->publishVersion($frDraft->versionInfo);
+        $updatedContent = $contentService->loadContent($frContent->id, null, $frContent->versionInfo->versionNo);
+        $this->assertEquals(
+            [
+                'eng-US' => 'Folder US',
+                'fre-FR' => 'Folder FR',
+            ],
+            $updatedContent->fields['name']
+        );
+
+        $dePublished = $contentService->publishVersion($deDraft->versionInfo);
+        $this->assertEquals(
+            [
+                'eng-US' => 'Folder US',
+                'ger-DE' => 'Folder GER',
+                'fre-FR' => 'Folder FR',
+            ],
+            $dePublished->fields['name']
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $newDraft
+     * @param string $identifier
+     * @param \eZ\Publish\Core\FieldType\TextLine\Value $newValue
+     * @param string $languageCode
+     *
+     * @return \eZ\Publish\Core\Repository\Values\Content\ContentUpdateStruct
+     */
+    protected function translateField(Content $newDraft, string $identifier, Value $newValue, string $languageCode): ContentUpdateStruct
+    {
+        $contentUpdateData = new ContentUpdateStruct([
+            'initialLanguageCode' => $languageCode,
+            'fields' => $newDraft->getFields(),
+        ]);
+
+        $nameFieldDefinition = array_filter(
+            $contentUpdateData->fields,
+            function (Field $field) use ($identifier) {
+                return $identifier === $field->fieldDefIdentifier;
+            }
+        );
+
+        $contentUpdateData->fields[key($nameFieldDefinition)] = new Field(
```

Comment:
You cannot use here Field Definition identifier as key. What if you want to update single field with two translations? Anyway, as stated somewhere above, this is not the way to update fields.